packages/rps/src/core/app-data.ts

```diff
@@ -1,26 +1,73 @@
 import {Weapon} from './weapons';
+import {BigNumber, defaultAbiCoder, bigNumberify, keccak256} from 'ethers/utils';
+import {HashZero} from 'ethers/constants';
+import {randomHex} from '../utils/randomHex';
 
-export interface Resting {
-  type: 'resting';
+export enum PositionType {
+  Start, // 0
+  RoundProposed, // 1
+  RoundAccepted, // 2
+  Reveal, // 3
 }
+export interface RPSData {
+  positionType: PositionType;
+  stake: BigNumber; // uint256
+  preCommit: string; // bytes32
+  playerAWeapon: Weapon;
+  playerBWeapon: Weapon;
+  salt: string; // bytes32
+}
+export interface Start {
+  type: 'start';
+}
+
+export type RoundProposed = Pick<RPSData, 'stake' | 'preCommit'> & {type: 'roundProposed'};
```

Comment:
There's a lot more cognitive overhead to processing this line, than if you had written:
```ts
export type RoundProposed {
  type: 'roundProposed';
  stake: BigNumber;
  preCommit: string;
}
```
Would you be averse to replacing it?

Reply:
That's true  / I would not be.  I think my way is slightly safer, since it's harder for me to mistype the properties. But I can just make sure I don't mistype...